core/include/vecmem/utils/tuple.hpp

```diff
@@ -93,6 +93,25 @@ struct tuple<T, Ts...> {
     VECMEM_HOST_AND_DEVICE constexpr tuple(U &&head, tuple<Us...> &&tail)
         : m_head(std::forward<U>(head)), m_tail(std::move(tail)) {}
 
+    /// Assignment operator from a (slightly) different tuple type
+    ///
+    /// @param parent The parent to copy
+    ///
+    template <typename U, typename... Us,
+              std::enable_if_t<vecmem::details::negation<std::is_same<
+                                   tuple<T, Ts...>, tuple<U, Us...>>>::value &&
+                                   sizeof...(Ts) == sizeof...(Us),
+                               bool> = true>
+    VECMEM_HOST_AND_DEVICE constexpr tuple &operator=(
+        const tuple<U, Us...> &parent) {
+        if (static_cast<const void *>(this) !=
+            static_cast<const void *>(&parent)) {
```

Comment:
Now that I think about it, since this function now makes sure that the auto-generated assignment operator was left in place, this `if(...)` check is not actually needed...